connectors/memory/memory.go

```diff
@@ -486,8 +435,15 @@ func (c *Connector) Shutdown() error {
 	return nil
 }
 
+// NewConnector creates a new in-memory connector
+func NewConnector() *Connector {
```

Comment:
Beautiful. I really like this. One centralized place for initializing the Connector.

Quick Question, instead of returning a `*Connector`, would it make sense to just return a `dosa.Connector`. That way the struct isn't tightly coupled?